api_tests/preprints/views/test_preprint_institutions_relationship.py

```diff
@@ -0,0 +1,304 @@
+        """
+        Test that read users cannot remove institutions.
+        """
+        preprint.affiliated_institutions.add(institution_A)
+        preprint.save()
+        update_institutions_payload = {'data': []}
+        res = app.put_json_api(url, update_institutions_payload, auth=read_user_with_institutional_affiliation.auth, expect_errors=True)
+        assert res.status_code == 403
+
+    def test_update_affiliated_institutions_remove_write_user(self, app, write_user_with_institutional_affiliation, preprint, url, institution_A):
+        """
+        Test that write users cannot remove institutions.
+        """
+        preprint.affiliated_institutions.add(institution_A)
+        preprint.save()
+        update_institutions_payload = {'data': []}
+        res = app.put_json_api(url, update_institutions_payload, auth=write_user_with_institutional_affiliation.auth, expect_errors=True)
+        assert res.status_code == 200
+
+    def test_update_affiliated_institutions_remove_admin_without_affiliation(self, app, admin_without_institutional_affiliation, preprint, url, institution_A):
+        """
+        Test that admins without affiliation can remove institutions.
+        """
+        preprint.affiliated_institutions.add(institution_A)
+        preprint.save()
+        update_institutions_payload = {'data': []}
+        res = app.put_json_api(url, update_institutions_payload, auth=admin_without_institutional_affiliation.auth)
+        assert res.status_code == 200
+
+    def test_update_affiliated_institutions_remove_admin_with_affiliation(self, app, admin_with_institutional_affiliation, preprint, url, institution_A):
+        """
+        Test that admins with affiliation can remove institutions.
+        """
+        preprint.affiliated_institutions.add(institution_A)
+        preprint.save()
+        update_institutions_payload = {'data': []}
+        res = app.put_json_api(url, update_institutions_payload, auth=admin_with_institutional_affiliation.auth)
+        assert res.status_code == 200
+        preprint.reload()
+        assert institution_A not in preprint.affiliated_institutions.all()
+
+        log = preprint.logs.latest()
+        assert log.action == 'affiliated_institution_removed'
+        assert log.params['institution'] == {'id': institution_A._id, 'name': institution_A.name}
+
+    def test_preprint_institutions_list_get_unauthenticated(self, app, url):
+        """
+        Test that unauthenticated users cannot retrieve the list of affiliated institutions for a preprint.
+        """
+        res = app.get(url, expect_errors=True)
+        assert res.status_code == 200
+
+    def test_preprint_institutions_list_get_no_permissions(self, app, user, url):
+        """
+        Test that users without permissions cannot retrieve the list of affiliated institutions for a preprint.
+        """
+        res = app.get(url, auth=user.auth, expect_errors=True)
```

Comment:
Why are we expecting errors here?

Reply:
Same, this test is incorrect. GET to the preprint-institution relationship endpoint should be public.